### ISSUE TITLE
Skip to main link - interaction with cookie banner

### ISSUE THREAD

**Issue by msayoung** [localgovdrupal/localgov_base#354]
@Mjeffbrown reports that the skip to main link is missing on some content types. 

Directories and Guide pages. 

I need to test and confirm whether this is base theme or the submodules.

**Comment by Mjeffbrown** [comment]
Pages affected include:

Guide - https://new.newcastle.gov.uk/births-deaths-marriages-ceremonies/registration-certificates/order-copy-certificate

Directory venue - https://new.newcastle.gov.uk/births-deaths-marriages-ceremonies/directory-approved-ceremony-venues

**Comment by andybroomfield** [comment]
For what its worth, I'm seeing it on those pages, and clicking it does take me to the page title.

<img width="1396" alt="Screenshot 2023-01-26 at 2 23 13 pm" src="https://user-images.githubusercontent.com/1467480/214860229-9e69dc0c-5db3-4085-a40e-5e2c905ef4a7.png">
<img width="1396" alt="Screenshot 2023-01-26 at 2 23 22 pm" src="https://user-images.githubusercontent.com/1467480/214860425-a9b16d33-b81d-444d-ba64-a05af3426db2.png">

**Comment by keelanfh** [comment]
I can see it on those pages too. When you click tab, the focus state does go to the cookie control first, but the skip link is next.

**Comment by markconroy** [comment]
> For what its worth, I'm seeing it on those pages, and clicking it does take me to the page title.

Same for me.

**Comment by Mjeffbrown** [comment]
Thanks. I'm doing a forced re-check on SiteImprove across those pages to see if it continues to show an issue. Will report back, but fingers crossed it was just a temporary aberration.

**Comment by markconroy** [comment]
Siteimprove might be reporting a false positive or something like that. Probably best to manually test it as well.

**Comment by andybroomfield** [comment]
Also on the demo site which suggests it is in the default install.

https://demo.localgovdrupal.org/national-parks-uk
and
https://demo.localgovdrupal.org/adult-health-and-social-care/travel-passes-and-support/blue-badges

<img width="1368" alt="Screenshot 2023-01-27 at 2 48 44 pm" src="https://user-images.githubusercontent.com/1467480/215115458-115a31e5-8879-4da9-80f6-552aa66a8401.png">
<img width="1368" alt="Screenshot 2023-01-27 at 2 49 32 pm" src="https://user-images.githubusercontent.com/1467480/215115490-fb74abd4-2155-478a-90b4-3d79f1ca228a.png">

**Comment by keelanfh** [comment]
I've just checked in the Siteimprove browser extension, and it also flags for me. But it doesn't flag on https://demo.localgovdrupal.org/national-parks-uk.

The message from Siteimprove states:

> The first item reached by keyboard should be a link to the main content of the page. This link should be visible when receiving focus from the keyboard.

I wonder if the issue is not related to LocalGov Drupal, but to the cookie control banner that you have. The cookie banner is the first item reached by keyboard, so a keyboard user would have to interact with, or dismiss, the cookie banner, before reaching the Skip link. So that wouldn't meet Siteimprove's check definition.

**Comment by markconroy** [comment]
I think the first clickable item _should_ be the cookie banner thing so you can give informed consent about what you want to do on the website. 

This sounds like a false positive from Siteimprove and maybe the issue should be logged with them to update their API.

**Comment by keelanfh** [comment]
(Slightly off-topic but) my overall preference (and our planned approach at Essex) is to not require people to interact with the cookie banner - and so I'd prefer to have the Skip link come first. We don't require mouse users to interact with a cookie banner, so wouldn't require keyboard users to do so either. This is consistent with https://www.gov.uk/. Naturally we'll only set non-strictly necessary cookies if someone explicitly opts in.

On sites where you require people to interact with the banner (e.g. https://www.blog.gov.uk/), I guess it makes a bit more sense. I still think you could have the skip link appearing first, and default to not dropping any cookies. You do tend to see a higher % of users accepting cookies if you require them to interact with a banner.

Having said that, I don't think this is a technical WCAG 2.1 AA failure and it is a bit of a false positive from Siteimprove.

**Comment by markconroy** [comment]
If people do not interact with the banner, I presume it pops up on every page?

**Comment by keelanfh** [comment]
Yes, that's right.

**Comment by markconroy** [comment]
It's a really tricky one. On the one hand you should not need to interact with the cookie banner and you should still be able to read all content. Implication being: the cookie banner is the last clickable item. But ... if the cookie banner is annoying you and you want to click "Reject all cookies" to get rid of it, you need to tab through _every_ item on the page.

On the other hand, if you put it at the top of the page, then you can interact with the content directly and need to tab through the cookie banner every time you load a new page.

I have no solution to this issue to be honest.

**Comment by andybroomfield** [comment]
I think this is a known bug with cookie banner from EU Cookie Compliance module and possibly other cookie banners in general that put the banner at the top. Because it injects itself in JS to be right below the body tag. 
That script would have to be modified to account for the skip link, or theme will need to script for trying to detect it, both seem impractical to ship with the module / theme.

**Comment by andybroomfield** [comment]
For reference, gov.uk goes above.
<img width="1397" alt="Screenshot 2023-05-09 at 10 11 03 am" src="https://github.com/localgovdrupal/localgov_base/assets/1467480/f92797ac-a901-4e46-baba-6b67b7f23b4a">

BHCC has now implemented a change detailed in https://github.com/localgovdrupal/localgov_eu_cookie_compliance/issues/13 that also moves it to the top.

<img width="1388" alt="Screenshot 2023-05-16 at 1 19 26 pm" src="https://github.com/localgovdrupal/localgov_base/assets/1467480/d6f0062c-ce7c-49f1-afe1-e4b389ed9e48">

**Comment by keelanfh** [comment]
Closing this as not directly LGD related - https://github.com/localgovdrupal/localgov_eu_cookie_compliance/issues/13 can be used for discussion of the LGD cookie banner.
